Modules/Module_Import/NII2BIDS/xASL_bids_BIDSifyASLJSON.m

```diff
@@ -526,9 +526,15 @@
 ASLContextM0Index = cellfun(@(x)~isempty(x),ASLContextM0Index); % Create a vector out of it
 ASLContextM0Index = ASLContextM0Index(1:dimASL(4)); % Remove the last empty field
 
+% If ASLContext contains M0-scans, and repetitionTimePreparation vector is a scalar, and RepetitionTimeM0 is defined, then we create a repetitionTimePreparation vector
```

Comment:
No I would always create a vector, for all parameters for all DICOMs. And only if it is unique/the same for all, we use it in the old way, otherwise we create a vector in BIDS. Would that not be more general (less new surprises for us)?